pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2599,10 +2583,14 @@ static __always_inline int capture_file_write(struct pt_regs *ctx, u32 event_id)
             bin_args.full_size = io_vec.iov_len;
         }
     }
-    bpf_map_update_elem(&bin_args_map, &id, &bin_args, BPF_ANY);
 
-    // Send file data
-    bpf_tail_call(ctx, &prog_array, TAIL_SEND_BIN);
+    if (p.event->buf_off < ARGS_BUF_SIZE - sizeof(bin_args_t)) {
```

Comment:
I see this logic repeating itself throughout the PR, we should probably make it a util function

Reply:
I see what you mean but I don't think it is a good idea to hide a tail call behind a function and anyway its only 3 lines of code, so I prefer keeping it as is

Reply:
I think we could name it something like `tail_call_send_bin` and it would make the side effect explicit in it's name, but I also understand the caution taken.
Perhaps a macro would fit here better, something like a `send_bin_probe`, which would work like the `BPF_KPROBE` macro but embed this logic in addition?

Reply:
done